editor/plugins/canvas_item_editor_plugin.cpp

```diff
@@ -3575,7 +3575,7 @@ void CanvasItemEditor::_draw_invisible_nodes_positions(Node *p_node, const Trans
 		_draw_invisible_nodes_positions(p_node->get_child(i), parent_xform, canvas_xform);
 	}
 
-	if (ci && !ci->_edit_use_rect() && (!editor_selection->is_selected(ci) || _is_node_locked(ci))) {
+	if (ci && !ci->_edit_use_rect() && (!editor_selection->is_selected(ci) || _is_node_locked(ci)) && show_position_gizmos) {
```

Comment:
```suggestion
	if (show_position_gizmos && ci && !ci->_edit_use_rect() && (!editor_selection->is_selected(ci) || _is_node_locked(ci))) {
```
Cheapest condition first.

Reply:
Do you want to have the other 2 checks reordered as well?

Line 3714 and 3720
`if (p_node->has_meta("_edit_lock_") && show_lock_gizmos) {`
to
`if (show_lock_gizmos && p_node->has_meta("_edit_lock_")) {`

Reply:
Yes